src/Compilers/VisualBasic/Portable/Symbols/Tuples/TupleMethodSymbol.vb

```diff
@@ -0,0 +1,144 @@
+                Return Me._underlyingMethod
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property AssociatedSymbol() As Symbol
+            Get
+                Return Me._containingType.GetTupleMemberSymbolForUnderlyingMember(Of Symbol)(Me._underlyingMethod.ConstructedFrom.AssociatedSymbol)
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property ContainingSymbol() As Symbol
+            Get
+                Return Me._containingType
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property ExplicitInterfaceImplementations() As ImmutableArray(Of MethodSymbol)
+            Get
+                Return Me._underlyingMethod.ConstructedFrom.ExplicitInterfaceImplementations
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property Parameters() As ImmutableArray(Of ParameterSymbol)
+            Get
+                Dim isDefault As Boolean = Me._lazyParameters.IsDefault
+                If isDefault Then
+                    InterlockedOperations.Initialize(Of ParameterSymbol)(Me._lazyParameters, Me.CreateParameters())
+                End If
+                Return Me._lazyParameters
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property IsSub() As Boolean
+            Get
+                Return Me._underlyingMethod.IsSub
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property ReturnType() As TypeSymbol
+            Get
+                Return Me._underlyingMethod.ReturnType
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property ReturnTypeCustomModifiers() As ImmutableArray(Of CustomModifier)
+            Get
+                Return Me._underlyingMethod.ReturnTypeCustomModifiers
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property TypeArguments() As ImmutableArray(Of TypeSymbol)
+            Get
+                Return StaticCast(Of TypeSymbol).From(Me._typeParameters)
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property TypeParameters() As ImmutableArray(Of TypeParameterSymbol)
+            Get
+                Return Me._typeParameters
+            End Get
+        End Property
+
+        Public Sub New(container As TupleTypeSymbol, underlyingMethod As MethodSymbol)
+            Debug.Assert(underlyingMethod.ConstructedFrom Is underlyingMethod)
+            Me._containingType = container
+            Me._underlyingMethod = underlyingMethod
+
+            ' PROTOTYPE: tuples Should alpha rename typeparameters in case a tuple method has any.
+            Me._typeParameters = Me._underlyingMethod.TypeParameters
+        End Sub
+
+        Private Function CreateParameters() As ImmutableArray(Of ParameterSymbol)
+            Return Me._underlyingMethod.Parameters.SelectAsArray(Of ParameterSymbol)(Function(p) New TupleParameterSymbol(Me, p))
```

Comment:
Consider using `_underlyingMethod.Parameters.SelectAsArray(Function(p, m) New TupleParameterSymbol(m, p), Me)` instead, to allow the delegate to be shared across instances.

Also, the C# versions of these methods can use `SelectAsArray` as well.